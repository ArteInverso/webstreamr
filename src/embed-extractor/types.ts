import { StreamWithMeta } from '../types';

export interface EmbedExtractor {
  readonly id: string;

  readonly label: string;

  readonly extract: (embedUrl: string, language: string) => Promise<StreamWithMeta>;
}
