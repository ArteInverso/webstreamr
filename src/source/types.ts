import { ContentType } from 'stremio-addon-sdk';
import { Context, CountryCode } from '../types';
import { Id } from '../utils';

export interface SourceResult {
  countryCode: CountryCode;
  title?: string;
  url: URL;
}

export interface Source {
  readonly id: string;

  readonly label: string;

  readonly contentTypes: ContentType[];

  readonly countryCodes: CountryCode[];

  handle(ctx: Context, type: ContentType, id: Id): Promise<(SourceResult[])>;
}
