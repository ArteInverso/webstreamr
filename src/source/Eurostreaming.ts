import { ContentType } from 'stremio-addon-sdk';
import * as cheerio from 'cheerio';
import { Source, SourceResult } from './types';
import { Fetcher, getImdbId, Id, ImdbId } from '../utils';
import { Context, CountryCode } from '../types';

export class Eurostreaming implements Source {
  public readonly id = 'eurostreaming';

  public readonly label = 'Eurostreaming';

  public readonly contentTypes: ContentType[] = ['series'];

  public readonly countryCodes: CountryCode[] = [CountryCode.it];

  private readonly fetcher: Fetcher;

  public constructor(fetcher: Fetcher) {
    this.fetcher = fetcher;
  }

  public readonly handle = async (ctx: Context, _type: string, id: Id): Promise<SourceResult[]> => {
    const imdbId = await getImdbId(ctx, this.fetcher, id);

    const seriesPageUrl = await this.fetchSeriesPageUrl(ctx, imdbId);
    if (!seriesPageUrl) {
      return [];
    }

    const html = await this.fetcher.text(ctx, seriesPageUrl);

    const $ = cheerio.load(html);

    const mainDataLinkElements = $(`[data-num="${imdbId.season}x${imdbId.episode}"][data-link!="#"]`);
    const mirrorDataLinkElements = $(`[data-num="${imdbId.season}x${imdbId.episode}"]`)
      .siblings('.mirrors')
      .children('[data-link!="#"]');

    const title = $('meta[property="og:title"]').attr('content') as string;

    return Promise.all(mainDataLinkElements
      .add(mirrorDataLinkElements)
      .map((_i, el) => new URL(($(el).attr('data-link') as string).replace(/^(https:)?\/\//, 'https://')))
      .toArray()
      .filter(url => !url.host.match(/eurostreaming/))
      .map(url => ({ countryCode: CountryCode.it, referer: seriesPageUrl, title: `${title.trim()} ${imdbId.season}x${imdbId.episode}`, url })),
    );
  };

  private fetchSeriesPageUrl = async (ctx: Context, imdbId: ImdbId): Promise<URL | undefined> => {
    const html = await this.fetcher.textPost(
      ctx,
      new URL('https://eurostreaming.my/index.php'),
      JSON.stringify({
        do: 'search',
        subaction: 'search',
        search_start: 0,
        full_search: 0,
        result_from: 1,
        story: imdbId.id,
      }),
      {
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded',
        },
      },
    );

    const $ = cheerio.load(html);

    const url = $('.post-content a[href]:first')
      .map((_i, el) => $(el).attr('href'))
      .get(0);

    return url !== undefined ? new URL(url) : url;
  };
}
