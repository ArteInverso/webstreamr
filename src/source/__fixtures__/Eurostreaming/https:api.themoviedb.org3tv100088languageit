{"adult":false,"backdrop_path":"/uQ4lG7E7mlyKsGvbASftQ6Hu2IX.jpg","created_by":[{"id":1295692,"credit_id":"5e84f03598f1f10016a985c0","name":"Neil Druckmann","original_name":"Neil Druckmann","gender":2,"profile_path":"/bVUsM4aYiHbeSYE1xAw2H5Z1ANU.jpg"},{"id":35796,"credit_id":"5e84f06a3344c600153f6a57","name":"Craig Mazin","original_name":"Craig Mazin","gender":2,"profile_path":"/uEhna6qcMuyU5TP7irpTUZ2ZsZc.jpg"}],"episode_run_time":[],"first_air_date":"2023-01-15","genres":[{"id":18,"name":"Dramma"}],"homepage":"https://www.hbo.com/the-last-of-us","id":100088,"in_production":true,"languages":["en"],"last_air_date":"2025-05-25","last_episode_to_air":{"id":5994273,"name":"Convergenza","overview":"Nel mezzo della battaglia tra le fazioni in guerra di Seattle, la ricerca di Ellie la porta verso uno scontro devastante.","vote_average":6.1,"vote_count":51,"air_date":"2025-05-25","episode_number":7,"episode_type":"finale","production_code":"","runtime":50,"season_number":2,"show_id":100088,"still_path":"/4IZqCZRmBq9LifHFQkYA1QqqE0x.jpg"},"name":"The Last of Us","next_episode_to_air":null,"networks":[{"id":49,"logo_path":"/tuomPhY2UtuPTqqFnKMVHvSb724.png","name":"HBO","origin_country":"US"}],"number_of_episodes":16,"number_of_seasons":2,"origin_country":["US"],"original_language":"en","original_name":"The Last of Us","overview":"Vent’anni dopo la distruzione della civiltà moderna Joel, uno scaltro sopravvissuto, viene incaricato di far uscire Ellie, una ragazza di 14 anni, da una zona di quarantena sotto stretta sorveglianza. Un compito all’apparenza facile che si trasforma presto in un viaggio brutale e straziante, in cui i due si troveranno a dover attraversare gli Stati Uniti insieme e a dipendere l’uno dall’altra per sopravvivere.","popularity":89.0703,"poster_path":"/6WlffOd3bszyndGgsKGcUL6k28n.jpg","production_companies":[{"id":125281,"logo_path":"/3hV8pyxzAJgEjiSYVv1WZ0ZYayp.png","name":"PlayStation Productions","origin_country":"US"},{"id":11073,"logo_path":"/aCbASRcI1MI7DXjPbSW9Fcv9uGR.png","name":"Sony Pictures Television","origin_country":"US"},{"id":23217,"logo_path":"/kXBZdQigEf6QiTLzo6TFLAa7jKD.png","name":"Naughty Dog","origin_country":"US"},{"id":119645,"logo_path":null,"name":"Word Games","origin_country":"US"},{"id":115241,"logo_path":null,"name":"The Mighty Mint","origin_country":"US"},{"id":3268,"logo_path":"/tuomPhY2UtuPTqqFnKMVHvSb724.png","name":"HBO","origin_country":"US"}],"production_countries":[{"iso_3166_1":"US","name":"United States of America"}],"seasons":[{"air_date":"2023-01-15","episode_count":9,"id":144593,"name":"Stagione 1","overview":"Da Boston a Salt Lake City, attraverso un mondo devastato da una piaga apocalittica: un uomo e una ragazza viaggiano per tentare di salvare l'umanità.","poster_path":"/bSWFzaboUFxfxBE1pIWlRsp5TvE.jpg","season_number":1,"vote_average":7.7},{"air_date":"2025-04-12","episode_count":7,"id":405376,"name":"Stagione 2","overview":"Cinque anni dopo gli eventi della prima stagione, Joel ed Ellie saranno trascinati in un conflitto fra di loro e contro un mondo persino più pericoloso e imprevedibile di quello che si erano lasciati alle spalle.","poster_path":"/2IXXPbfqGX6HUHD6HX2z56g2ra.jpg","season_number":2,"vote_average":6.4}],"spoken_languages":[{"english_name":"English","iso_639_1":"en","name":"English"}],"status":"Returning Series","tagline":"Quando ti perdi nell'oscurità, cerca la luce","type":"Scripted","vote_average":8.507,"vote_count":6307}