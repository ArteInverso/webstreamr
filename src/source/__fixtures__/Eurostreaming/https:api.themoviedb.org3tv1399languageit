{"adult":false,"backdrop_path":"/2OMB0ynKlyIenMJWI2Dy9IWT4c.jpg","created_by":[{"id":9813,"credit_id":"5256c8c219c2956ff604858a","name":"David Benioff","original_name":"David Benioff","gender":2,"profile_path":"/bOlW8pymCeQLfwPIvc2D1MRcUoF.jpg"},{"id":228068,"credit_id":"552e611e9251413fea000901","name":"D. B. Weiss","original_name":"D. B. Weiss","gender":2,"profile_path":"/2RMejaT793U9KRk2IEbFfteQntE.jpg"}],"episode_run_time":[],"first_air_date":"2011-04-17","genres":[{"id":10765,"name":"Sci-Fi & Fantasy"},{"id":18,"name":"Dramma"},{"id":10759,"name":"Action & Adventure"}],"homepage":"https://www.hbo.com/game-of-thrones","id":1399,"in_production":false,"languages":["en"],"last_air_date":"2019-05-19","last_episode_to_air":{"id":1551830,"name":"Il Trono di Spade","overview":"Il destino dei Sette Regni è a rischio, mentre il capitolo finale del Trono di Spade viene scritto.","vote_average":4.562,"vote_count":361,"air_date":"2019-05-19","episode_number":6,"episode_type":"finale","production_code":"806","runtime":80,"season_number":8,"show_id":1399,"still_path":"/zBi2O5EJfgTS6Ae0HdAYLm9o2nf.jpg"},"name":"Il Trono di Spade","next_episode_to_air":null,"networks":[{"id":49,"logo_path":"/tuomPhY2UtuPTqqFnKMVHvSb724.png","name":"HBO","origin_country":"US"}],"number_of_episodes":73,"number_of_seasons":8,"origin_country":["US"],"original_language":"en","original_name":"Game of Thrones","overview":"Nell'immaginario mondo di Westeros, in un tempo dal sapore medievale, s'intrecciano le vicende di alcune casate che lottano per la supremazia: i Baratheon, i Lannister, gli Stark, i Targaryen sono solo alcune delle famiglie che, tra guerre manifeste e intrighi clandestini, in un interminabile rovescio di fortune e alleanze, cercano di trionfare sugli avversari e prendere possesso del Trono di Spade, simbolo del potere e del comando. Tratto dal ciclo di romanzi Cronache del ghiaccio e del fuoco di George R. R. Martin.","popularity":327.2332,"poster_path":"/n4IxkaZanXQU89WNRxt8h1fw1yb.jpg","production_companies":[{"id":76043,"logo_path":"/9RO2vbQ67otPrBLXCaC8UMp3Qat.png","name":"Revolution Sun Studios","origin_country":"US"},{"id":12525,"logo_path":null,"name":"Television 360","origin_country":""},{"id":5820,"logo_path":null,"name":"Generator Entertainment","origin_country":"GB"},{"id":12526,"logo_path":null,"name":"Bighead Littlehead","origin_country":"US"}],"production_countries":[{"iso_3166_1":"GB","name":"United Kingdom"},{"iso_3166_1":"US","name":"United States of America"}],"seasons":[{"air_date":"2010-12-05","episode_count":286,"id":3627,"name":"Speciali","overview":"","poster_path":"/aos6lC1JGYt6ZRL85lgstNsfSeY.jpg","season_number":0,"vote_average":0.0},{"air_date":"2011-04-17","episode_count":10,"id":3624,"name":"Stagione 1","overview":"Il Re dei Sette Regni di Westeros, Robert Baratheon, si reca a Grande Inverno, per chiedere al suo amico Ned Stark di accettare la carica di Primo Cavaliere. Inoltre il Re propone un matrimonio tra suo figlio Joffrey, erede al trono di spade e la figlia di Ned, Sansa. Da questo incontro prende il via una serie di eventi che porterà Ned, Sansa e Arya ad Approdo del Re, mentre il resto della famiglia Stark rimane a Grande Inverno per controllare il territorio. I delicati equilibri politici si infrangono quando qualcuno cerca di far tacere definitivamente il piccolo Bran Stark, colpevole di aver scoperto un terribile segreto. Lady Catelyn Stark riceve conferma dei suoi sospetti e accusa i Lannister del misfatto, evento scatenante che darà il via a una serie di conflitti. Intanto al di là del Mare Stretto, Viserys Targaryen, figlio del Re Folle Aerys II, deposto da Robert Baratheon, dà sua sorella Daenerys in sposa a Khal Drogo, un potente condottiero che gli promette un esercito per riappropriarsi del trono dei Sette Regni.","poster_path":"/r6Qoa10PBMP6oc7bo8qjQbM8oPQ.jpg","season_number":1,"vote_average":8.4},{"air_date":"2012-04-01","episode_count":10,"id":3625,"name":"Stagione 2","overview":"La guerra tra le famiglie dei Sette Regni è ormai giunta. Mentre Tyrion Lannister esercita le funzioni di Primo Cavaliere e regge a stento il Trono di Spade, su cui è seduto il folle giovane re Joffrey, altri uomini pretendono di essere regnanti: Robb Stark, che ha radunato i vessilli del Nord; Stannis Baratheon, la cui rigida lucidità è in parte annebbiata da un nuovo culto monoteista; Renly Baratheon, con il supporto delle armate dell'Altopiano; Balon Greyjoy, con le sue navi lunghe; Daenerys Targaryen, la giovane principessa esiliata, sopravvissuta alle fiamme e madre dei draghi, che cerca un esercito al di là del mare. Nel frattempo, oltre la Barriera, Jon Snow e i Guardiani della notte si preparano all'impatto con forze oscure e dimenticate.","poster_path":"/brNW7pFKBAsokrEkW3wNuY75LQ8.jpg","season_number":2,"vote_average":8.3},{"air_date":"2013-03-31","episode_count":10,"id":3626,"name":"Stagione 3","overview":"Jon Snow è tra i Bruti e mentre Daenerys Targaryen cerca di ottenere un esercito nelle terre di Astapor, la situazione ad Approdo del Re cambia: i Tyrell acquistano più potere e cercano di entrare nelle grazie dei potenti Lannister. In viaggio verso nord, Bran Stark vive sogni sempre più realistici e incontra un nuovo amico che lo accompagnerà nel suo viaggio verso la Barriera. Jaime Lannister e Brienne di Tarth continuano la loro marcia verso sud con non poche difficoltà, mentre Tyrion e Cersei Lannister non sono più padroni di loro stessi dall'arrivo del loro padre nella capitale. Arya Stark incontra il sacerdote Thoros di Myr e si unisce alla Fratellanza Senza Vessilli. Rifugiato alla Roccia del Drago, Stannis Baratheon attende un segnale dal Signore della Luce, mentre la Sacerdotessa Rossa è in viaggio alla ricerca del figlio bastardo nelle cui vene scorre il sangue di un Re. Theon Greyjoy è stato rapito dal bastardo di Roose Bolton, un individuo di raro sadismo che lo tormenterà alternando torture, mutilazioni e false speranze. Robb Stark sente il peso della guerra e cerca di dare uno scopo ai suoi uomini, ma intrighi e tradimenti lo aspettano presso coloro che credeva suoi alleati.","poster_path":"/dGK0t2FluacFOPaWYi6m8UXQV7b.jpg","season_number":3,"vote_average":8.3},{"air_date":"2014-04-06","episode_count":10,"id":3628,"name":"Stagione 4","overview":"Rassicurati dall'esito del massacro delle “Nozze di Sangue” nella Terra dei Fiumi, durante il quale sono stati eliminati molti membri della famiglia rivale degli Stark, i Lannister sembrano essere solidamente in possesso del Trono di Spade… ma minacce più insidiose si apprestano a sorgere dal loro stesso egoismo e dall'arrivo di nuovi nemici da Sud, da Nord e da Est. Mentre Stannis Baratheon prosegue nella ricostituzione del suo esercito alla Roccia del Drago, un nuovo pericolo arriva da Sud, quando Oberyn Martell, la “Vipera Rossa di Dorne” odiata dai Lannister, giunge ad Approdo del Re per presenziare alle nozze di Joffrey con Margaery Tyrell, e forse per mettere in atto un piano segreto. A Nord gli esausti Guardiani della Notte sembrano essere in numero insufficiente per affrontare l'inesorabile avanzata verso la Barriera dell'esercito dei Bruti guidati da Mance Rayder, sulle cui tracce si trova peraltro un nemico ancor più pericoloso: gli Estranei, i non-morti. Daenerys Targaryen, accompagnata dai suoi temibili tre draghi e dal suo esercito di Immacolati, è pronta a “liberare” Meereen, la più grande città orientale della Baia degli Schiavi, che potrebbe procurarle quelle forze imponenti di cui necessita per raggiungere il suo obiettivo finale: navigare verso il Continente Occidentale e rivendicare il Trono di Spade, appartenente alla sua famiglia e ora usurpato dai Lannister. ","poster_path":"/g4yHhoKBRQCb5p3fvjCadWQA07Z.jpg","season_number":4,"vote_average":8.5},{"air_date":"2015-04-12","episode_count":10,"id":62090,"name":"Stagione 5","overview":"Dopo la morte di Tywin, capofamiglia Lannister, per mano di Tyrion che sfugge alla condanna e salpa per Essos, ad Approdo del Re la Regina Madre Cersei Lannister tenta di mantenere solido il potere della Corona, ora guidata dal giovane Tommen Baratheon, che vede minacciato dall'ingerenza dei Tyrell. Cersei conferisce dunque potere alla figura enigmatica e austera dell'Alto Passero e dal suo ordine di fedeli ai Sette Dei, ma il piano le si ritorcerà contro. Jaime Lannister parte per una pericolosa missione a Dorne, mentre suo fratello Tyrion vaga per il Continente Orientale in cerca di uno scopo, fino a quando non incontrerà Daenerys. Petyr Baelish fa sposare Sansa Stark a Ramsay Bolton per permetterle di tornare a Grande Inverno. Conscio dell'imminente arrivo della Lunga Notte, Jon Snow tenta di fondare un'alleanza con i Bruti, ma le sue decisioni sono poco amate tra i Guardiani della Notte. Nel frattempo, Re Stannis Baratheon si prepara alla conquista dei Sette Regni, anche se la sua marcia verso Grande Inverno si rivela piena di ostacoli. Ad aspettarlo nel cuore del Nord vi sono Roose e Ramsay Bolton, decisi a difendere il potere ottenuto con sangue e tradimento. A Meereen, Daenerys Targaryen affronta le conseguenze della sua politica senza il consiglio di ser Jorah Mormont, il quale cerca in tutti i modi di riguadagnare la fiducia della Madre dei Draghi. Arrivata a Braavos, Arya Stark aspira ad acquisire i poteri degli Uomini Senza Volto e diventare un'assassina.","poster_path":"/ps43O8NaAstF9Y4Kz9T8HZfJy7u.jpg","season_number":5,"vote_average":8.2},{"air_date":"2016-04-24","episode_count":10,"id":71881,"name":"Stagione 6","overview":"Ad Approdo del Re, la Regina Madre Cersei Lannister e suo fratello Jaime tentano di riunire la famiglia reale, ancora minacciata dal Credo Militante dell'Alto Passero. A Grande Inverno, Ramsay Bolton cerca di convincere le grandi famiglie del Nord ad allearsi con lui, mentre a Braavos Arya Stark prosegue il suo percorso per diventare \"nessuno\". A Dorne, Ellaria Sand è pronta a prendere le redini del principato con l'aiuto delle Serpi delle Sabbie. Petyr Baelish, ottenuto il completo controllo sul giovane Robin Arryn e l'esercito della Valle, deve riconquistare la fiducia di Sansa Stark. Nelle Terre dei Fiumi, mentre la Fratellanza Senza Vessilli si imbatte in un personaggio dimenticato, continua la guerra fra la Casa Frey e la Casa Tully. Al Castello Nero, Ser Davos Seaworth e Lady Melisandre affrontano le insidie sorte dopo la morte di Jon Snow, mentre oltre la Barriera Bran Stark prosegue il suo allenamento con il Corvo con tre occhi. Theon Greyjoy lascia Sansa nelle mani di Brienne di Tarth per tornare dalla sua famiglia. Ignari del suo stato, i Greyjoy di Pyke continuano ad aspirare alla grandezza. Mentre Daenerys Targaryen è prigioniera di Khal Moro e attende che la fortuna torni dalla sua parte, Tyrion Lannister collabora con Varys per tentare di controllare la città di Meereen.","poster_path":"/j8VM9xG9i9LvSl278PXMqnTpK4I.jpg","season_number":6,"vote_average":8.3},{"air_date":"2017-07-16","episode_count":7,"id":81266,"name":"Stagione 7","overview":"Ad Approdo del Re, dopo essersi liberata da tutti i suoi nemici nella capitale e aver perso anche l'ultimo dei suoi figli, la nuova Regina dei Sette Regni Cersei Lannister – affiancata da suo fratello Jaime, Qyburn e Ser Gregor Clegane – valuta un'alleanza con il nuovo Re delle Isole di Ferro Euron Greyjoy e Casa Tarly. A Grande Inverno, il Re del Nord Jon Snow – consigliato da Sansa Stark e Davos Seaworth – si prepara allo scontro con il Re della Notte e la sua armata di Estranei e non-morti, mentre gli giunge notizia dell'arrivo a Roccia del Drago di Daenerys Targaryen, supportata dagli Immacolati, i Dothraki, Tyrion Lannister, Ellaria Sand, Yara e Theon Greyjoy, Olenna Tyrell, Varys e Melisandre di Asshai. Nel frattempo, Arya Stark – dopo aver sterminato la famiglia Frey – cerca di tornare dalla sua famiglia, mentre Sandor Clegane si dirige al Nord con la Fratellanza Senza Vessilli. Bran Stark – ora diventato il Corvo con tre occhi – giunge finalmente al Castello Nero e viene accolto da Edd Tollett e i Guardiani della Notte. A Vecchia Città, Samwell Tarly aiuta Jon Snow inviandogli informazioni preziose sugli Estranei trovate alla Cittadella, dove risiede anche Jorah Mormont, in cerca di una cura per il morbo grigio.","poster_path":"/2ytnoSWacZ2o6Ng4wDDNqWfqEAJ.jpg","season_number":7,"vote_average":8.2},{"air_date":"2019-04-14","episode_count":6,"id":107971,"name":"Stagione 8","overview":"La Grande Guerra è arrivata, il Muro è caduto e l'esercito dei morti del Re della Notte marcia inesorabilmente verso Westeros. La fine è vicina, ma chi salirà sul Trono di Spade?","poster_path":"/bw2pwDo1liQlYcTReAcG9WM6G9U.jpg","season_number":8,"vote_average":6.2}],"spoken_languages":[{"english_name":"English","iso_639_1":"en","name":"English"}],"status":"Ended","tagline":"L'inverno sta arrivando.","type":"Scripted","vote_average":8.456,"vote_count":25268}