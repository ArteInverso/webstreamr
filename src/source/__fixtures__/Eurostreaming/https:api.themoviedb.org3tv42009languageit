{"adult":false,"backdrop_path":"/dg3OindVAGZBjlT3xYKqIAdukPL.jpg","created_by":[{"id":211845,"credit_id":"52595fb5760ee346619587b5","name":"Charlie Brooker","original_name":"Charlie Brooker","gender":2,"profile_path":"/dmdLd5I4cnpCZzoITm26NOwu6Rn.jpg"}],"episode_run_time":[],"first_air_date":"2011-12-04","genres":[{"id":10765,"name":"Sci-Fi & Fantasy"},{"id":18,"name":"Dramma"},{"id":9648,"name":"Mistero"}],"homepage":"https://www.netflix.com/title/70264888","id":42009,"in_production":true,"languages":["en"],"last_air_date":"2025-04-10","last_episode_to_air":{"id":6085099,"name":"USS Callister: Infinity","overview":"Nanette Cole e l'equipaggio della USS Callister sono bloccati in un vasto universo virtuale dove lottano per la sopravvivenza contro trenta milioni di giocatori.","vote_average":7.5,"vote_count":48,"air_date":"2025-04-10","episode_number":6,"episode_type":"finale","production_code":"","runtime":91,"season_number":7,"show_id":42009,"still_path":"/6JxoU4B1VMYf2tt8VPpNl8juKwm.jpg"},"name":"Black Mirror","next_episode_to_air":null,"networks":[{"id":26,"logo_path":"/ws2U3Ftuxkt5C6pUY0QtkEfuPiH.png","name":"Channel 4","origin_country":"GB"},{"id":213,"logo_path":"/wwemzKWzjKYJFfCeiB57q3r4Bcm.png","name":"Netflix","origin_country":""}],"number_of_episodes":32,"number_of_seasons":7,"origin_country":["GB"],"original_language":"en","original_name":"Black Mirror","overview":"Una serie antologica di racconti angosciosi e spiazzanti che rivelano i tratti umani più oscuri, le innovazioni più sensazionali e tanto altro ancora.","popularity":107.8447,"poster_path":"/seN6rRfN0I6n8iDXjlSMk1QjNcq.jpg","production_companies":[{"id":76757,"logo_path":null,"name":"House of Tomorrow","origin_country":"GB"},{"id":18663,"logo_path":"/khiCshsZBdtUUYOr4VLoCtuqCEq.png","name":"Zeppotron","origin_country":"GB"},{"id":133905,"logo_path":"/tEEIQIYHQgPyOBgvoAlqPXfpEPX.png","name":"Broke and Bones","origin_country":"GB"}],"production_countries":[{"iso_3166_1":"GB","name":"United Kingdom"}],"seasons":[{"air_date":"2014-12-16","episode_count":1,"id":76596,"name":"Speciali","overview":"","poster_path":"/zvfw0f35QmfHCPq5qvdHquFjrQC.jpg","season_number":0,"vote_average":0.0},{"air_date":"2011-12-04","episode_count":3,"id":51964,"name":"Stagione 1","overview":"Questa miniserie antologica immagina un futuro fantascientifico dove le persone pedalano per guadagnare \"meriti\" e vengono dotate di un impianto di memoria.","poster_path":"/ztdy1f1cAjwsO0GpVKzNuqZ0PMv.jpg","season_number":1,"vote_average":7.7},{"air_date":"2013-02-11","episode_count":3,"id":51965,"name":"Stagione 2","overview":"Tra i temi della seconda stagione di questa serie antologica spiccano la dipendenza dai social media e le implicazioni dei reality dal vivo.","poster_path":"/y71DeJiAv0dV8H8hiFnuIuyc0Gx.jpg","season_number":2,"vote_average":7.1},{"air_date":"2016-10-21","episode_count":6,"id":63871,"name":"Stagione 3","overview":"Nella terza stagione, un paradiso surreale per gli innamorati, un giovane tormentato da strani messaggi e un'app che esercita un potere inquietante.","poster_path":"/3mKYrlZpFbpFu7CaVxoleO68MFG.jpg","season_number":3,"vote_average":7.7},{"air_date":"2017-12-29","episode_count":6,"id":96276,"name":"Stagione 4","overview":"Una fantasia fuori controllo, oggetti che svelano oscuri segreti e una donna inseguita da un implacabile cacciatore. Tutto questo in nuove storie di tecnologia impazzita.","poster_path":"/rQqrNbXeFsSdL3CF2zBVqymnsHL.jpg","season_number":4,"vote_average":7.3},{"air_date":"2019-06-05","episode_count":3,"id":124320,"name":"Stagione 5","overview":"Un videogioco trasforma una lunga amicizia, una società di social media subisce un sequestro e una ragazza crea un legame con la versione robot del suo idolo pop.","poster_path":"/5xjf5aLeooPYWqH66vB80jMJT9u.jpg","season_number":5,"vote_average":6.5},{"air_date":"2023-06-15","episode_count":5,"id":331809,"name":"Stagione 6","overview":"Cinque nuovi episodi tra antichi misteri, paure del futuro, tecnologia nemica e un'inquietante rappresentazione del mondo.","poster_path":"/kwATNh8JC1IJclnVwk8x7wZb2b4.jpg","season_number":6,"vote_average":6.4},{"air_date":"2025-04-10","episode_count":6,"id":383778,"name":"Stagione 7","overview":"","poster_path":"/aRN9LmPQHe8c1B2eR7MB3tiWFWk.jpg","season_number":7,"vote_average":7.1}],"spoken_languages":[{"english_name":"English","iso_639_1":"en","name":"English"}],"status":"Returning Series","tagline":"","type":"Scripted","vote_average":8.286,"vote_count":5587}