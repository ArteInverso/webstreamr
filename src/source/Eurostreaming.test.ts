import { Eurostreaming } from './Eurostreaming';
import { Fetcher, ImdbId } from '../utils';
import { Context } from '../types';
jest.mock('../utils/Fetcher');

// @ts-expect-error No constructor args needed
const fetcher = new Fetcher();
const ctx: Context = { id: 'id', ip: '127.0.0.1', config: { it: 'on' } };

describe('Eurostreaming', () => {
  let handler: Eurostreaming;

  beforeEach(() => {
    handler = new Eurostreaming(fetcher);
  });

  test('handles non-existent series gracefully', async () => {
    const streams = await handler.handle(ctx, 'series', new ImdbId('tt12345678', 1, 1));
    expect(streams).toHaveLength(0);
  });

  test('handle imdb black mirror s2e4', async () => {
    const streams = await handler.handle(ctx, 'series', new ImdbId('tt2085059', 2, 4));
    expect(streams).toMatchSnapshot();
  });
});
