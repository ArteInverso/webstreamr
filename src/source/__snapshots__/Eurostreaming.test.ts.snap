// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Eurostreaming handle imdb black mirror s2e4 1`] = `
[
  {
    "countryCode": "it",
    "referer": "https://eurostreaming.my/stream/284-black-mirror-streaming-streaming.html",
    "title": "Black Mirror 2x4",
    "url": "https://supervideo.cc/embed-anf0clp2vb6y.html",
  },
  {
    "countryCode": "it",
    "referer": "https://eurostreaming.my/stream/284-black-mirror-streaming-streaming.html",
    "title": "Black Mirror 2x4",
    "url": "https://supervideo.cc/embed-anf0clp2vb6y.html",
  },
  {
    "countryCode": "it",
    "referer": "https://eurostreaming.my/stream/284-black-mirror-streaming-streaming.html",
    "title": "Black Mirror 2x4",
    "url": "https://dropload.io/embed-978xyyxi3ldq.html",
  },
]
`;
